client/helpers.go

```diff
@@ -302,6 +302,19 @@ func IsAWSError(err error, code ...string) bool {
 	return false
 }
 
+func IgnoreCustomError(err error, code string, messageRegex string) bool {
+	var ae smithy.APIError
+	if !errors.As(err, &ae) {
+		return false
+	}
+
+	r, _ := regexp.Compile(messageRegex)
```

Comment:
r would be nil if supplied regex wasn't valid... Better to get `*regexp.Regexp` and leave the regex validation to be early (and much faster)